src/mapex/write_parsed_mappings.py

```diff
@@ -204,69 +209,82 @@ def write_parsed_mappings_stix(parsed_mappings, filepath):
             stix_object["id"]
             for stix_object in stix_bundle["objects"]
             if stix_object.get("name") == mapping["capability_id"]
-        ][0]
+        ]
+        related_target_ref = technique_target_dict.get(mapping["attack_object_id"], "")
 
         # account for None mapping_type on not_mappable items
         mapping_type = (
             mapping["mapping_type"].replace("_", "-")
             if mapping["mapping_type"]
             else None
         )
-        stix_bundle["objects"].append(
-            {
-                "type": "relationship",
-                "id": f"relationship--{relationship_uuid}",
-                "spec_version": "2.1",
-                "created": datetime.now().isoformat(),
-                "modified": datetime.now().isoformat(),
-                "relationship_type": mapping_type,
-                "source_ref": related_source_ref,
-                "target_ref": technique_target_dict.get(
-                    mapping["attack_object_id"], ""
-                ),
-            },
+        # do not add a relationship node for a non-mappable technique
+        if (
+            related_source_ref
+            and related_target_ref
+            and not mapping_type == "non-mappable"
```

Comment:
Let's avoid referencing `mapping_type` completely. We'll simply avoid creating relationships if either the source or target goes undefined.